Bugzilla/Search/Saved.pm

```diff
@@ -235,16 +291,24 @@ sub edit_link {
   return $self->{edit_link};
 }
 
+sub _whine_subjects {
+  my ($self) = @_;
+
+  return $self->{_whine_subjects} if exists $self->{_whine_subjects};
+
+  $self->{_whine_subjects} = Bugzilla->dbh->selectcol_arrayref(
+    'SELECT DISTINCT whine_events.subject FROM whine_events
+      INNER JOIN whine_queries ON whine_queries.eventid = whine_events.id
+      WHERE whine_events.owner_userid = ? AND whine_queries.query_name = ?',
+    undef, Bugzilla->user->id, $self->name);
+
+  return $self->{_whine_subjects};
+}
+
 sub used_in_whine {
   my ($self) = @_;
-  return $self->{used_in_whine} if exists $self->{used_in_whine};
-  ($self->{used_in_whine}) = Bugzilla->dbh->selectrow_array(
-    'SELECT 1 FROM whine_events INNER JOIN whine_queries
-                       ON whine_events.id = whine_queries.eventid
-          WHERE whine_events.owner_userid = ? AND query_name = ?', undef,
-    $self->{userid}, $self->name
-  ) || 0;
-  return $self->{used_in_whine};
+
+  return scalar(@{$self->_whine_subjects}) ? 1 : 0;
```

Comment:
```suggestion
  return @{$self->_whine_subjects} ? 1 : 0;
```

`scalar` is only needed when you're inside list. It's a bit like the opposite of the spread operator. Every time you see a `@expr` it's like `...var` in javascript, and `scalar` removes that (implicit) spreading.